File: 정글(Python)/정글_1_9663.py
#n 퀸
n=int(input())

pos = [0]*n
flag = [False]*n
flag_L = [False]*(2*n-1)
flag_R = [False]*(2*n-1)

def put():
    for i in range(n):
         print(pos[i],end="")
    print()

def set(count,i):
    for j in range(n):
        if not flag[j] and not flag_L[i+j] and not flag_R[n-1+i-j]:
            pos[i]=j
            if i == n - 1:
                # put()
                count +=1
            else:
                flag[j] = flag_L[i+j] = flag_R[n-1+i-j]=True
                count = set(count,i+1)
                flag[j] = flag_L[i+j] = flag_R[n-1+i-j]=False
    return count
count=0;
print(set(count,0))

File: Python/1주차_정렬,재귀/정글_1_9663.cpp
// 퀸
#include <bits/stdc++.h>
using namespace std;

int* pos;
int* flag;
int* flag_L;
int* flag_R;

void showqueen(int n){
    for(int i; i <n;i++){
        cout<<pos[i];
    } 
    cout<<endl;
}

void setqueen(int n, int count,int i){
    for(int j; j<n;j++){
        pos[i] =j;
        if (i == n-1){
            showqueen(n);
        }else{
            setqueen(n,count,i+1);
        }
    }
}

int main(void) {
    cin.tie(0);
    ios::sync_with_stdio(0);
    int n;
    cin >> n;

    flag = (int*)malloc(sizeof(int)*n);
    flag_L = (int*)malloc(sizeof(int)*(2*n-1));
    flag_R = (int*)malloc(sizeof(int)*(2*n-1));
    int count = 0;
    setqueen(n,count,0);

    return 0;
}
